lib/iceberg/events/self_submit_orders.js

```diff
@@ -15,12 +15,19 @@ const generateOrders = require('../util/generate_orders')
  */
 const onSelfSubmitOrders = async (instance = {}) => {
   const { state = {}, h = {} } = instance
-  const { emit } = h
+  const { emit, timeout, updateState } = h
   const { args = {}, gid } = state
   const { submitDelay } = args
-  const orders = generateOrders(state)
 
-  return emit('exec:order:submit:all', gid, orders, submitDelay)
+  if (submitDelay) {
+    const [id, t] = timeout(submitDelay)
```

Comment:
timeout needs to get cleared on `life_stop.js`?

Reply:
something alike

https://github.com/bitfinexcom/bfx-hf-algo/blob/55db8d21ec2e6e4868e7c444a54a8228e7e809bd/lib/twap/events/life_stop.js#L12-L20

?